x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/PEMKeyConfig.java

```diff
@@ -120,6 +122,8 @@ private static PrivateKey readPrivateKey(String keyPath, SecureString keyPasswor
             throw unreadableKeyConfigFile(accessException, KEY_FILE, key);
         } catch (AccessControlException securityException) {
             throw blockedKeyConfigFile(securityException, environment, KEY_FILE, key);
+        } catch (GeneralSecurityException e) {
+            throw new IllegalStateException("Error parsing Private Key from: " + keyPath, e);
```

Comment:
The X-Pack PemUtils would throw this `IllegalStateException` on `GeneralSecurityException` but the ssl-config version throws `GeneralSecurityException`.  
This isn't the nicest way to handle it, but my plan is to get rid of this class (PEMKeyConfig) entirely, so I opted for the smallest possible change.